EventFilter/L1TRawToDigi/plugins/implementations_stage2/CaloLayer1Setup.cc

```diff
@@ -0,0 +1,102 @@
+#include "FWCore/Framework/interface/stream/EDProducerBase.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "EventFilter/L1TRawToDigi/plugins/PackerFactory.h"
+#include "EventFilter/L1TRawToDigi/plugins/PackingSetupFactory.h"
+#include "EventFilter/L1TRawToDigi/plugins/UnpackerFactory.h"
+
+#include "CaloLayer1Setup.h"
+
+namespace l1t {
+   namespace stage2 {
+      std::unique_ptr<PackerTokens>
+      CaloLayer1Setup::registerConsumes(const edm::ParameterSet& cfg, edm::ConsumesCollector& cc)
+      {
+         return std::unique_ptr<PackerTokens>(new CaloLayer1Tokens(cfg, cc));
+      }
+
+      void
+      CaloLayer1Setup::fillDescription(edm::ParameterSetDescription& desc)
+      {
+         desc.addOptional<edm::InputTag>("ecalDigis");
+         desc.addOptional<edm::InputTag>("hcalDigis");
+         desc.addOptional<edm::InputTag>("caloRegions");
+      }
+
+      PackerMap
+      CaloLayer1Setup::getPackers(int fed, unsigned int fw)
+      {
+         PackerMap res;
+
+         if (fed == 1354) {
```

Comment:
rather than hardwiring these lists of packers,I think they should be config parameters. That way the code does not need to have hardwired fed numbers. [you already have "fed" in the config - just replace it with a list of numbers to use here -  [ [2,3],[3,4].....]